GroovyParsingException when using Map literal notation in gradle dependency with spaces

## What version of OpenRewrite are you using?
I am using

- OpenRewrite latest version

## How are you running OpenRewrite?
I am running a unit test to test some other issue fix

## What is the smallest, simplest way to reproduce the problem?
```groovy
plugins {
    id 'java'
}

repositories {
    mavenCentral()
}

dependencies {
    implementation( [group: "org.openrewrite", name: "rewrite-java", version: "latest.release"] )
}
```

## What did you expect to see?
Successful parsing of the groovy gradle file

## What did you see instead?
An error during parsing

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
java.lang.AssertionError: Failed to parse sources or run recipe

	at org.openrewrite.test.RewriteTest.lambda$defaultExecutionContext$14(RewriteTest.java:641)
	at org.openrewrite.groovy.GroovyParser.lambda$parseInputs$3(GroovyParser.java:158)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4853)
	at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4861)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:282)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:130)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:125)
	at org.openrewrite.gradle.GradleParserTest.dependencyNotations(GradleParserTest.java:193)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: org.openrewrite.groovy.GroovyParsingException: Failed to parse build.gradle, cursor position likely inaccurate.
	at org.openrewrite.groovy.GroovyParserVisitor.visit(GroovyParserVisitor.java:205)
	at org.openrewrite.groovy.GroovyParser.lambda$parseInputs$3(GroovyParser.java:147)
	... 24 more
Caused by: java.lang.StringIndexOutOfBoundsException: begin 190, end 190, length 178
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4606)
	at java.base/java.lang.String.substring(String.java:2709)
	at org.openrewrite.groovy.GroovyParserVisitor.whitespace(GroovyParserVisitor.java:2303)
	at org.openrewrite.groovy.GroovyParserVisitor.access$000(GroovyParserVisitor.java:82)
	at org.openrewrite.groovy.GroovyParserVisitor$RewriteGroovyVisitor.visitArgumentlistExpression(GroovyParserVisitor.java:752)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.visit(ArgumentListExpression.java:73)
	at org.openrewrite.groovy.GroovyParserVisitor$RewriteGroovyVisitor.visit(GroovyParserVisitor.java:620)
	at org.openrewrite.groovy.GroovyParserVisitor$RewriteGroovyVisitor.lambda$visitMethodCallExpression$10(GroovyParserVisitor.java:1688)
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
Sure

--- a/rewrite-gradle/src/test/java/org/openrewrite/gradle/GradleParserTest.java
+++ b/rewrite-gradle/src/test/java/org/openrewrite/gradle/GradleParserTest.java
@@ -189,6 +189,64 @@ class GradleParserTest implements RewriteTest {
     }
 
     @Test
+    void dependencyNotations() {
+        rewriteRun(
+          buildGradle(
+            """
+              plugins {
+                  id 'java'
+              }
+              
+              repositories {
+                  mavenCentral()
+              }
+              
+              dependencies {
+              
+                  // String notation
+                  implementation "org.openrewrite:rewrite-java:latest.release"
+                  implementation ("org.openrewrite:rewrite-java:latest.release")
+                  implementation ("org.openrewrite:rewrite-java:latest.release") { transitive = false }
+                  implementation ("org.openrewrite:rewrite-java:latest.release") {
+                      transitive = false
+                  }
+                  implementation ( "org.openrewrite:rewrite-java:latest.release" )
+                  implementation ( "org.openrewrite:rewrite-java:latest.release" ) { transitive = false }
+                  implementation ( "org.openrewrite:rewrite-java:latest.release" ) {
+                      transitive = false
+                  }
+              
+                  // Map notation
+                  implementation group: "org.openrewrite", name: "rewrite-java", version: "latest.release"
+                  implementation(group: "org.openrewrite", name: "rewrite-java", version: "latest.release")
+                  implementation(group: "org.openrewrite", name: "rewrite-java", version: "latest.release") { transitive = false }
+                  implementation(group: "org.openrewrite", name: "rewrite-java", version: "latest.release") {
+                      transitive = false
+                  }
+                  implementation( group: "org.openrewrite", name: "rewrite-java", version: "latest.release" )
+                  implementation( group: "org.openrewrite", name: "rewrite-java", version: "latest.release" ) { transitive = false }
+                  implementation( group: "org.openrewrite", name: "rewrite-java", version: "latest.release" ) {
+                      transitive = false
+                  }
+                  
+                  // Map literal notation
+                  implementation([group: "org.openrewrite", name: "rewrite-java", version: "latest.release"])
+                  implementation([group: "org.openrewrite", name: "rewrite-java", version: "latest.release"]) { transitive = false }
+                  implementation([group: "org.openrewrite", name: "rewrite-java", version: "latest.release"]) {
+                      transitive = false
+                  }
+                  implementation( [group: "org.openrewrite", name: "rewrite-java", version: "latest.release"] )
+                  implementation( [group: "org.openrewrite", name: "rewrite-java", version: "latest.release"] ) { transitive = false }
+                  implementation( [group: "org.openrewrite", name: "rewrite-java", version: "latest.release"] ) {
+                      transitive = false
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
     void kotlinDsl() {
         rewriteRun(
           buildGradleKts(

--- a/rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java
+++ b/rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java
@@ -719,7 +719,6 @@ public class GroovyParserVisitor {
                 // If it is wrapped in "[]" then this isn't a named arguments situation, and we should not lift the parameters out of the enclosing MapExpression
                 saveCursor = cursor;
                 whitespace();
-                cursor = saveCursor;
                 if ('[' != source.charAt(cursor)) {
                     // Bring named parameters out of their containing MapExpression so that they can be parsed correctly
                     MapExpression namedArgExpressions = (MapExpression) unparsedArgs.get(0);
@@ -729,6 +728,7 @@ public class GroovyParserVisitor {
                                             unparsedArgs.subList(1, unparsedArgs.size()).stream())
                                     .collect(toList());
                 }
+                cursor = saveCursor;
             }
 
             if (unparsedArgs.isEmpty()) {

